injection/sharedmain/main.go

```diff
@@ -340,6 +339,8 @@ func MainWithConfig(ctx context.Context, component string, cfg *rest.Config, cto
 	// returns an error.
 	<-egCtx.Done()
 
+	pprof.Shutdown(context.Background())
```

Comment:
should we have some kind of timeout on this context to ensure that we stop eventually here?

Reply:
Ah okay 👍